Google.Api.Gax.Grpc/ChannelPool.cs

```diff
@@ -22,12 +22,10 @@ namespace Google.Api.Gax.Grpc
     /// </summary>
     public sealed class ChannelPool
```

Comment:
I was doing exactly the same exercise today (trying to make gax.grpc a managed library). What I found is that ChannelPool and ClientBuilderBase are not used anywhere except tests. My naive implementation was to introduce 2 additional libraries. One will be called ` Google.Api.Gax.Grpc.Api` (by the analogy of how Grpc.Core was split) which will pool all managed code from the `Google.Api.Gax.Grpc`, another one is Google.Api.Gax.Grpc.AspNetCore which will contain managed client code.